### PR TITLE
More Homebrew/linuxbrew-core maintainer guide sections

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] ~Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).~
- [x] ~Have you successfully run `brew style` with your changes locally?~
- [x] ~Have you successfully run `brew tests` with your changes locally?~

-----

- Fix a bug where we were using the non-existent `$GITHUB_USER` envvar
  instead of `$HOMEBREW_GITHUB_USER` which exists and is used
  elsewhere in the Homebrew codebase.
- Document the bug where merge PRs don't go green in CI.
- Document how and when we pull bottles and use `brew pull --bottle`
  and `brew squash-bottle-pr`, and also how to recover when you
  accidentally publish a bottle.

This is probably easier to review commit by commit?

### CODE REVIEW

File: docs/Homebrew-linuxbrew-core-Maintainer-Guide.md [hunk 1]
```diff
@@ -245,6 +250,59 @@ for Linuxbrew` to the top of the formula, pushes the branch to GitHub
...
+as `brew pull --bottle` adds `Closes` and `Signed-off-by` to the
+commit message body.
+
+```bash
+lbrew-pull-bottle <PR-NUMBER> && brew squash-bottle-pr
+```
+
+For PRs where there have been force pushes or extra commits to fix the
+build or fix bottling syntax, we can't `brew squash-bottle-pr` as we
+must keep the fixes.
```
Comment (issyl0) [inline docs/Homebrew-linuxbrew-core-Maintainer-Guide.md]:
```suggestion
must keep the fixes. If the "# Build a bottle for Linuxbrew" comment is still in the formula, delete it.
```

File: docs/Homebrew-linuxbrew-core-Maintainer-Guide.md [hunk 2]
```diff
@@ -245,6 +250,59 @@ for Linuxbrew` to the top of the formula, pushes the branch to GitHub
...
+bottle". This means that the PR shows up as "closed" to the user, but
+they still get authorship credit. This is done with the following
+command:
+
+```bash
+HOMEBREW_BOTTLE_DOMAIN=https://linuxbrew.bintray.com brew pull --bottle --bintray-org=linuxbrew --test-bot-user=LinuxbrewTestBot <PR-NUMBER>
+```
+
+It saves a lot of time to alias this in your shell config. One
+possible alias is `lbrew-pull-bottle`.
```
Comment (sjackman) [inline docs/Homebrew-linuxbrew-core-Maintainer-Guide.md]:
```suggestion
possible alias is `brew-pull-bottle`.
```

File: docs/Homebrew-linuxbrew-core-Maintainer-Guide.md [hunk 3]
```diff
@@ -245,6 +250,59 @@ for Linuxbrew` to the top of the formula, pushes the branch to GitHub
...
+build or fix bottling syntax, we can't `brew squash-bottle-pr` as we
+must keep the fixes.
+
+The `brew pull` command *publishes* the bottle to BinTray and verifies
+that the SHA in the formula and the SHA of the downloaded file match.
+To verify a bottle, the script downloads the bottle from BinTray - if
+you're on an unstable connection, this may take a while or even time
+out. Publishing the bottle means that it's available as the latest
+version for users to download, so we have to be quick with pushing the
+commits to `origin`.
```
Comment (sjackman) [inline docs/Homebrew-linuxbrew-core-Maintainer-Guide.md]: > Publishing the bottle means that it's available as the latest
	version for users to download, so we have to be quick with pushing the
	commits to `origin`.

I didn't follow this sentence. I don't know of a time sensitive reason to need to quickly push to `origin` following a `brew pull --bottle`.

Comment (issyl0) [reply docs/Homebrew-linuxbrew-core-Maintainer-Guide.md]: Maybe I chose my words wrong. I was going for: it’s so we don’t forget and leave bottles published without users able to use them. I’ll drop this sentence entirely.

File: docs/Homebrew-linuxbrew-core-Maintainer-Guide.md [hunk 4]
```diff
@@ -245,6 +250,59 @@ for Linuxbrew` to the top of the formula, pushes the branch to GitHub
...
+For PRs where there have been force pushes or extra commits to fix the
+build or fix bottling syntax, we can't `brew squash-bottle-pr` as we
+must keep the fixes.
+
+The `brew pull` command *publishes* the bottle to BinTray and verifies
+that the SHA in the formula and the SHA of the downloaded file match.
+To verify a bottle, the script downloads the bottle from BinTray - if
+you're on an unstable connection, this may take a while or even time
+out. Publishing the bottle means that it's available as the latest
+version for users to download, so we have to be quick with pushing the
```
Comment (issyl0) [inline docs/Homebrew-linuxbrew-core-Maintainer-Guide.md]:
```suggestion
version for users to download, so remember to push your commits
```

**Comment by issyl0** [comment]
Thanks, @MikeMcQuaid and @sjackman. I've made the requested changes.

**Comment by MikeMcQuaid** [comment]
thanks again @issyl0!

**Pull request merged**